### DISCUSSION TITLE
Maximal area subdivision

### DISCUSSION THREAD

**Discussion by evallabres** [libigl/libigl#1954] [Q&A]
Impressive library you have here! 

I am working on growing shapes, and I need to impose a maximal area for the triangle faces. Is there any subdivision function in the library that just splits the triangles that exceed a maximal size? I could not seem to find it. If this is the case, I will implement it myself and added to the library if you think it is useful. 

I really appreciate any help and ideas you can provide.

Eva

**Comment by alecjacobson** [comment]
Currently this is not implemented. It wouldn't be too hard to add. For example, could follow how I've done it in https://github.com/alecjacobson/gptoolbox/blob/master/mesh/upsample.m This requires 1) implementing upsampling on a selected list of triangles (or _not_ upsampling those in a unselected list) and 2) calling repeatedly with the selection based on area. In my matlab implementation, I find it convenient to use a function handle that gets called every iteration. This saves the caller some logic.

**Reply by evallabres** [reply]
Thanks for your reply. Your comments were extremely useful! 

I implemented it and realized that after subdividing a subset of the triangles, the lattice does not preserve the hexagonal structure, which is a requirement of my model. 

 I need a different type of subdivision scheme. I would like that it keeps the area of all triangles around a similar selected size, and it preserves a certain hexagonal structure in the lattice. Does anyone know a subdivision scheme that fulfills these requirements?

Thank you very much,

Eva
